### ISSUE TITLE
[pickers] Incorrect font size in DatePicker with responsiveFontSizes

### ISSUE THREAD

**Issue by cyrilchapon** [mui/mui-x#5465]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

When wrapping my Theme with `responsiveFontSizes()` (default config), the `DatePicker` selected month appears bigger.

```
// Augment mui Theme type with DatePicker
import type {} from '@mui/x-date-pickers/themeAugmentation'

import { createTheme, responsiveFontSizes } from '@mui/material/styles'
import { cyan } from '@mui/material/colors'

const theme = responsiveFontSizes(createTheme({
  palette: {
    primary: {
      main: cyan.A700
    }
  }
}))

export default theme
```

<img width="365" alt="Capture d’écran 2022-07-11 à 23 39 27" src="https://user-images.githubusercontent.com/10728426/178363584-364e0673-ef18-497f-92fe-01aec706e3ed.png">

### Expected behavior 🤔

I'd expect to behave the same desktop size as without the responsive util :

<img width="417" alt="Capture d’écran 2022-07-11 à 23 39 12" src="https://user-images.githubusercontent.com/10728426/178363970-e41215ac-7fcd-493f-9bf4-b78a36d3f01b.png">

### Steps to reproduce 🕹

Steps:

1. Wrap `createTheme` with `responsiveFontSizes`
2. Use `DatePicker` with at least `'month'` view

### Your environment 🌎

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  System:
    OS: macOS 12.2.1
  Binaries:
    Node: 16.14.0 - ~/.nvm/versions/node/v16.14.0/bin/node
    Yarn: 1.22.17 - ~/.nvm/versions/node/v16.14.0/bin/yarn
    npm: 8.3.1 - ~/.nvm/versions/node/v16.14.0/bin/npm
  Browsers:
    Chrome: 103.0.5060.114
    Edge: Not Found
    Firefox: 93.0
    Safari: 15.3
  npmPackages:
    @emotion/react: ^11.9.3 => 11.9.3 
    @emotion/styled: ^11.9.3 => 11.9.3 
    @mui/base:  5.0.0-alpha.88 
    @mui/icons-material: ^5.8.4 => 5.8.4 
    @mui/material: ^5.8.7 => 5.8.7 
    @mui/private-theming:  5.8.6 
    @mui/styled-engine:  5.8.7 
    @mui/system:  5.8.7 
    @mui/types:  7.1.4 
    @mui/utils:  5.8.6 
    @mui/x-date-pickers: ^5.0.0-beta.0 => 5.0.0-beta.0 
    @types/react: ^18.0.15 => 18.0.15 
    react: 18.2.0 => 18.2.0 
    react-dom: 18.2.0 => 18.2.0 
    typescript: 4.7.4 => 4.7.4 
```
</details>

(used Chrome)

**Comment by m4theushw** [comment]
Reproduction: https://codesandbox.io/s/materialuipickers-demo-mui-x-forked-7u34ke?file=/demo.tsx

Open the preview in a new window or increase it to simulate a desktop resolution.

---

The selected month uses the `h5` variant when selected, but the other months use `subtitle1`.

https://github.com/mui/mui-x/blob/59a809c1337a172a49ba0ad498f75c89db45b854/packages/x-date-pickers/src/MonthPicker/PickersMonth.tsx#L75

Only this should be already enough to display the selected month bigger, but this doesn't occur because we force all months to use the styles from `theme.typography.subtitle1`.

https://github.com/mui/mui-x/blob/59a809c1337a172a49ba0ad498f75c89db45b854/packages/x-date-pickers/src/MonthPicker/PickersMonth.tsx#L32

The bug only appears when `responsiveFontSizes` is used because the `h5` typography style is enriched to have font size variations depending on the resolution. The media queries added to enrich the styles come after and win over `theme.typography.subtitle1`, causing the selected month to be rendered with the font size from the `h5` variant, not from `subtitle1`. You don't see this problem on mobile because the media queries only kick in on desktop resolutions.

Since the usage of the `h5` variant has no much effect here, because `subtitle1` style is spread anyway, I believe we can unify the variant used:

```diff
diff --git a/packages/x-date-pickers/src/MonthPicker/PickersMonth.tsx b/packages/x-date-pickers/src/MonthPicker/PickersMonth.tsx
index dbd3e64d0..0c581042f 100644
--- a/packages/x-date-pickers/src/MonthPicker/PickersMonth.tsx
+++ b/packages/x-date-pickers/src/MonthPicker/PickersMonth.tsx
@@ -72,7 +72,7 @@ export const PickersMonth: React.FC<MonthProps> = (props) => {
       onClick={handleSelection}
       onKeyDown={onSpaceOrEnter(handleSelection)}
       color={selected ? 'primary' : undefined}
-      variant={selected ? 'h5' : 'subtitle1'}
+      variant="subtitle1"
       disabled={disabled}
       {...other}
     />
```

@flaviendelangle do you see any reason to keep `h5`?

**Comment by flaviendelangle** [comment]
In `PickersMonth` the core team went with an other approach which is to avoid using `Typography` altogether.

**Comment by flaviendelangle** [comment]
Fixed by #6034
